src/mlpack/core/optimizers/adam/nadam_update.hpp

```diff
@@ -0,0 +1,160 @@
+/**
+ * @file nadam_update.hpp
+ * @author Sourabh Varshney
+ *
+ * Nadam update rule. Nadam is an optimizer that combines the effect of Adam and 
+ * NAG to the gradient descent to improve its Performance.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_ADAM_NADAM_UPDATE_HPP
+#define MLPACK_CORE_OPTIMIZERS_ADAM_NADAM_UPDATE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Nadam is an optimizer that combines the Adam and NAG.
+ *
+ * For more information, see the following.
+ *
+ * @code
```

Comment:
Let's use:

```
* @code
* @techreport{Dozat2015,
*   title       = {Incorporating Nesterov momentum into Adam},
*   author      = {Timothy Dozat},
*   institution = {Stanford University},
*   address     = {Stanford},
*   year        = {2015}
* }
* @endcode
```

here.

Reply:
This was an ICLR paper, should we try and cite it like that instead?  I know this is a really pedantic comment so it is not a big deal either way.  And I also really dislike citing ICLR papers because I can never figure out how to do it because they don't *really* have proceedings, they just dump everything on arXiv or somewhere...

https://openreview.net/forum?id=OM0jvwB8jIp57ZJjtNEZ